subsys/usb/host/usbh_ch9.c

```diff
@@ -60,12 +54,11 @@ int usbh_req_setup(const struct device *dev,
 		if (usb_reqtype_is_to_device(&req) && data != NULL) {
```

Comment:
While this commit does not modify this particular condition, I think it might be good to assert (outside this function) on `(data != NULL && wLength) || (data == NULL && !wLength)`. Then we could remove the check against the `data != NULL) from here.

Reply:
Not sure where ASSERT should be placed, all the callers are simplified, I can imagine to add
`__ASSERT((buf != NULL && wLength) || (buf == NULL && !wLength), "Blabla");` just after memcpy().

Reply:
Just after memcpy() is fine.